packages/data-provider/src/utils/dictionarySchemaRelations.ts

```diff
@@ -0,0 +1,41 @@
+import { Dictionary } from '@overture-stack/lyric-data-model';
+import { SchemaDefinition } from '@overturebio-stack/lectern-client/lib/schema-entities.js';
+
+export interface SchemaNode {
+	schemaName: string;
+	fieldName: string;
+	parent?: SchemaNode;
+}
+
+/**
+ * Returns all the children and it's relations by each schema on a Dictionary
+ * @param {Dictionary} dictionary
+ * @returns {Record<string, SchemaNode[]>}
+ */
+export const getDictionarySchemaRelations = (dictionary: Dictionary): Record<string, SchemaNode[]> => {
+	const dictionaryRelations: Record<string, SchemaNode[]> = {};
+
+	dictionary.dictionary.map((schemaDefinition: SchemaDefinition) => {
+		const childrenSchemaName = schemaDefinition.name;
+
+		schemaDefinition.restrictions?.foreignKey?.map((foreignKey) => {
+			const parentSchemaName = foreignKey.schema;
+
+			foreignKey.mappings.map((mapping) => {
+				const childrenNode: SchemaNode = {
+					schemaName: childrenSchemaName,
+					fieldName: mapping.local,
+					parent: {
+						schemaName: parentSchemaName,
+						fieldName: mapping.foreign,
+					},
+				};
+
+				dictionaryRelations[parentSchemaName] = dictionaryRelations[parentSchemaName] || [];
+				dictionaryRelations[parentSchemaName].push(childrenNode);
+			});
+		});
+	});
```

Comment:
This `.map` is not returning anything, but looks to want to be building the contents of the object `dictionaryRelations`... This is a `reduce` operation.

If you want to leave it written this way, change the `.map` to a `.forEach`. Since you aren't actually returning anything there's no need to build the array that `.map` returns.

Reply:
refactoring this function as suggested